apps/cowswap-frontend/src/modules/tokens/hooks/useEnoughBalance.ts

```diff
@@ -75,22 +75,37 @@ export interface EnoughBalanceParams extends Omit<UseEnoughBalanceParams, 'check
  * @param params Parameters to check balance and optionally the allowance
  * @returns true if the account has enough balance (and allowance if it applies)
  */
-export function hasEnoughBalanceAndAllowance(params: EnoughBalanceParams): boolean {
+export function hasEnoughBalanceAndAllowance(params: EnoughBalanceParams): boolean | undefined {
   const { account, amount, balances, nativeBalance, allowances } = params
+
+  if (!account || !amount) {
+    return undefined
+  }
+
   const isNativeCurrency = amount?.currency.isNative
   const token = amount?.currency.wrapped
   const tokenAddress = getAddress(token)
 
-  const balance = tokenAddress ? balances[tokenAddress]?.value : undefined
-  const allowance = (tokenAddress && allowances && allowances[tokenAddress]?.value) || undefined
-
-  if (!account || !amount) {
-    return false
+  const enoughBalance = (() => {
+    const balance = tokenAddress ? balances[tokenAddress]?.value : undefined
+    const balanceAmount = isNativeCurrency ? nativeBalance : balance || undefined
+    return isEnoughAmount(amount, balanceAmount)
+  })()
```

Comment:
Do you need this inner fn declaration/execution?
If you do, how about a `useMemo`?

Reply:
I didn't add it because is not expensive operation as you pointed out. If you feel strong about it, i can add.
I only added this for readability and prevent logic errors (i felt it was a bit too much chained functions)

Reply:
16f15cded907718e96b438595529f125473f7066